dash/include/dash/algorithm/Generate.h

```diff
@@ -48,6 +48,55 @@ void generate (
   std::generate(lfirst, llast, gen);
 }
 
+/**
+ * Assigns each element in range [first, last) a value generated by the
+ * given function object g. The index passed to the function is
+ * a global index.
+ *
+ * Being a collaborative operation, each unit will invoke the given
+ * function on its local elements only.
+ *
+ * \tparam      ElementType    Type of the elements in the sequence
+ *                             invoke, deduced from parameter \c gen
+ * \tparam      UnaryFunction  Unary function with signature
+ *                             \c ElementType(index_t)
+ *
+ * \complexity  O(d) + O(nl), with \c d dimensions in the global iterators'
+ *              pattern and \c nl local elements within the global range
+ *
+ * \ingroup     DashAlgorithms
+ */
+template <
+    typename ElementType,
+    class    PatternType,
+    class    UnaryFunction >
+void generate_with_index(
+  /// Iterator to the initial position in the sequence
+  GlobIter<ElementType, PatternType> first,
+  /// Iterator to the final position in the sequence
+  GlobIter<ElementType, PatternType> last,
+  /// Generator function
+  UnaryFunction                      gen) {
+  /// Global iterators to local index range:
+  auto index_range  = dash::local_index_range(first, last);
```

Comment:
To be honest, I don't even know where to start looking. I guess I could crawl through a host of undocumented code, still not knowing where to start or how to put the pieces together...